proto/iavl_api.proto

```diff
@@ -0,0 +1,336 @@
+  // returning an error if the proof or key is invalid.
+  rpc VerifyAbsence(VerifyAbsenceRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      get: "/v1/range_proof/verify_absence"
+    };
+  }
+
+  // Rollback resets the working tree to the latest saved version, discarding
+  // any unsaved modifications.
+  rpc Rollback(google.protobuf.Empty) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      post: "/v1/rollback"
+      body: "*"
+    };
+  }
+
+  // Returns the committed versions
+  rpc GetAvailableVersions(google.protobuf.Empty) returns (GetAvailableVersionsResponse) {
+    option (google.api.http) = {
+      get: "/v1/available_versions"
+    };
+  }
+
+  // Load the most recent version
+  rpc Load(google.protobuf.Empty) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      post: "/v1/load"
+      body: "*"
+    };
+  }
+
+  // Load a specific version
+  rpc LoadVersion(LoadVersionRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      post: "/v1/load_version"
+      body: "*"
+    };
+  }
+
+  // Load a specific version and delete all the more recent versions
+  rpc LoadVersionForOverwriting(LoadVersionForOverwritingRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      post: "/v1/load_version_for_overwriting"
+      body: "*"
+    };
+  }
+
+  // Get the number of leaves in the tree
+  rpc Size(google.protobuf.Empty) returns (SizeResponse) {
+    option (google.api.http) = {
+      get: "/v1/size"
+    };
+  }
+
+  rpc List(ListElementsRequest) returns (stream ListElementsResponse) {
+    option (google.api.http) = {
+      get: "/v1/list"
+    };
+  }
+
+}
+
+// ----------------------------------------------------------------------------
+// Request types
+// ----------------------------------------------------------------------------
+
+message HasRequest {
+  bytes key = 1;
+}
+
+message HasVersionedRequest {
+  int64 version = 1;
+  bytes key = 2;
+}
+
+message GetRequest {
+  bytes key = 1;
+}
+
+message GetVersionedRequest {
+  int64 version = 1;
+  bytes key = 2;
+}
+
+message SetRequest {
+  bytes key = 1;
+  bytes value = 2;
+}
+
+message RemoveRequest {
+  bytes key = 1;
+}
+
+message DeleteVersionRequest {
+  int64 version = 1;
+}
+
+message VersionExistsRequest {
+  int64 version = 1;
+}
+
+message VerifyRequest {
+  bytes root_hash = 1;
+  iavl.RangeProof proof = 2;
+}
+
+message VerifyItemRequest {
+  bytes root_hash = 1;
+  iavl.RangeProof proof = 2;
+  bytes key = 3;
+  bytes value = 4;
+}
+
+message VerifyAbsenceRequest {
+  bytes root_hash = 1;
+  iavl.RangeProof proof = 2;
+  bytes key = 3;
+}
+
+message LoadVersionRequest {
+  int64 version = 1;
+}
+
+message LoadVersionForOverwritingRequest {
+  int64 version = 1;
+}
+
+message ListElementsRequest {
+  bytes from_key = 1;
+  bytes to_key = 2;
+  bool ascending = 3;
```

Comment:
Maybe we want to use `descending` instead, so that the default order when not specified is ascending. Or use an enum with the default set to ascending.